dataprofiler/tests/profilers/test_profiler_class_options.py

```diff
@@ -0,0 +1,111 @@
+    
+        # Treat is_enabled as a BooleanOption
+        for key in option.structured_options.properties:
+            expected_error = "type object 'structured_options.{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option.set({'structured_options.{}.is_enabled.is_enabled'.format(key): True})
+    
+    def test_validate_helper(self):
+        # Valid cases should return [] while invalid cases should return a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+
+        # Default Configuration Is Valid
+        self.assertEqual([], option._validate_helper())
+        
+        # Variable Path Is Not A String
+        expected_error = "The variable path must be a string."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            option._validate_helper(1)
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option._validate_helper())))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.structured_options = ProfilerOptions()
+
+        expected_error = set()
+        expected_error.add('{}.structured_options must be a StructuredOptions.'.format(optpth,))
+
+        self.assertSetEqual(expected_error, set(option._validate_helper()))
+            
+    def test_validate(self):
+        # Valid cases should return None while invalid cases should return or throw a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+    
+        # Default Configuration Is Valid
+        self.assertEqual(None, option.validate())
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option.validate(raise_error=False))))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.structured_options = ProfilerOptions()
+
+        expected_error = set()
+        expected_error.add('{}.structured_options must be a StructuredOptions.'.format(optpth,))
+        self.assertSetEqual(expected_error, set(option.validate(raise_error=False)))
```

Comment:
Again, see my other 2 comments about sets not being necessary for this case and raise_error being False not being as informative of a test case.